src/Symfony/Component/EventDispatcher/EventDispatcher.php

```diff
@@ -259,6 +265,41 @@ protected function doDispatch($listeners, $eventName, Event $event)
         }
     }
 
+    /**
+     * @param string|object $subscriber
+     * @param string|array  $params
+     *
+     * @return string
+     */
+    protected function inferEventName($subscriber, $params): string
```

Comment:
this should be in the pass, i.e. compile-time only

Reply:
Hum, but this would mean getSubscribedEvent wouldn't by that decoupled anymore as the pass would be a requirement to allow bypassing the event name.
And on the other side, I don't like having the reflection-related logic in the main object.
I think I'm :-1: on this idea sorry.

Reply:
> this should be in the pass, i.e. compile-time only

The logic is executed on compile-time only – in the full-stack framework scenario. If I moved the reflection logic into the pass, this feature would become a DI-only feature as it would not work anymore when using the dispatcher standalone.

Reply:
Yes, exactly what I meant also.
We just closed #30760 to save the code from doing more Reflection, even at compile time.
It would be consistent to close this one also, sorry again.

Reply:
All right.